src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -884,116 +925,125 @@ int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *
-                        os_calloc(1, sizeof(cv_scoring_system), report->cvss3);
-                        w_strdup(cvss3_vector, report->cvss3->vector_string);
-                        report->cvss3->base_score = strtod(cvss3, NULL);
-                    }
-                    if (!report->cwe) w_strdup(cwe, report->cwe);
 
-                    sqlite3_finalize(stmt);
-                    stmt = NULL;
-
-                    // Adding references URLs
-                    if (sqlite3_prepare_v2(db, vu_queries[VU_REFS_QUERY], -1, &stmt, NULL) != SQLITE_OK) {
-                        goto error;
-                    }
+            if (!report->alert_origin) os_strdup("Vendor feed", report->alert_origin);
 
-                    sqlite3_bind_text(stmt, 1, cve, -1, NULL);
-                    sqlite3_bind_text(stmt, 2, target, -1, NULL);
+            if (pkg->vuln_cond) {
+                if (!report->pending) report->pending = pkg->vuln_cond->pending;
+                if (!report->state) w_strdup(pkg->vuln_cond->state, report->state);
+                if (!report->operation) w_strdup(pkg->vuln_cond->operation, report->operation);
+                if (!report->operation_value) w_strdup(pkg->vuln_cond->operation_value, report->operation_value);
+                if (!report->condition) w_strdup(pkg->vuln_cond->condition, report->condition);
+            }
+            
+            // Adding OVALs information
+            if (sqlite3_prepare_v2(db, vu_queries[VU_GET_VULN_INFO], -1, &stmt, NULL) != SQLITE_OK) {
+                goto error;
+            }
 
-                    while (sql_result = wm_vuldet_step(stmt), sql_result == SQLITE_ROW) {
-                        reference = (char *)sqlite3_column_text(stmt, 0);
+            sqlite3_bind_text(stmt, 1, cve, -1, NULL);
+            sqlite3_bind_text(stmt, 2, target, -1, NULL);
+
+            if (sql_result = wm_vuldet_step(stmt), sql_result == SQLITE_ROW) {
+                title = (char *)sqlite3_column_text(stmt, 0);
+                severity = (char *)sqlite3_column_text(stmt, 1);
+                published = (char *)sqlite3_column_text(stmt, 2);
+                updated = (char *)sqlite3_column_text(stmt, 3);
+                rationale = (char *)sqlite3_column_text(stmt, 4);
+                cvss = (char *)sqlite3_column_text(stmt, 5);
+                cvss3 = (char *)sqlite3_column_text(stmt, 6);
+                cvss_vector = (char *)sqlite3_column_text(stmt, 7);
+                cvss3_vector = (char *)sqlite3_column_text(stmt, 8);
+                cwe = (char *)sqlite3_column_text(stmt, 9);
+
+                if (!report->title) w_strdup(title, report->title);
+                if (!report->rationale) w_strdup(rationale, report->rationale);
+                if (!report->severity) w_strdup(severity, report->severity);
+                if (!report->published) w_strdup(published, report->published);
+                if (!report->updated) w_strdup(updated ? updated : published, report->updated);
```

Comment:
If `updated` is not available, I would not add the field instead of reporting the published date twice

Reply:
DONE